backend/danswer/tools/headers.py

```diff
@@ -0,0 +1,18 @@
+from fastapi.datastructures import Headers
+
+from danswer.configs.model_configs import TOOL_PASS_THROUGH_HEADERS
+
+
+def get_tool_headers(headers: dict[str, str] | Headers) -> dict[str, str]:
+    if not TOOL_PASS_THROUGH_HEADERS:
+        return {}
+
+    pass_through_headers = {}
+    for key in TOOL_PASS_THROUGH_HEADERS:
+        lowercase_key = key.lower()
+        if key in headers:
+            pass_through_headers[key] = headers[key]
+        elif lowercase_key in headers:
+            pass_through_headers[lowercase_key] = headers[lowercase_key]
```

Comment:
Agree re: a docstring– believe that would help make this clear

Reply:
Understand the function is being used here for filtering tool headers, but it should be written more generically as an function of the form extract_headers(headers, TOOL_PASS_THROUGH_HEADERS) without accessing global state.  This allows the function to be pure and gives us a utility function we can put somewhere in case we want to reuse it.